turtle_graphics/turtle_graphics.js

```diff
@@ -0,0 +1,204 @@
+class Turtle {
+    constructor(x, y){        
+        this.x = x;
+        this.y = y;
+        this.way = 'east';
+        this.p = [[x,y]];    
+    }
+
+    forward(num){
+        if(this.way === 'east'){
+              for(let i = 1 ; i <= num; i++){
+              this.p.push([this.x + i, this.y])
+              }
+              this.x += num
+          } else if(this.way === 'south'){
+              for(let i = 1; i <= num; i++){
+              this.p.push([this.x, this.y + i])
+              }
+              this.y += num
+          } else if(this.way === 'west'){
+              for(let i = 1; i <= num; i++){
+              this.p.push([this.x - i, this.y])
+              }
+              this.x -= num
+          } else if(this.way === 'north'){
+              for(let i = 1; i <= num; i++){
+              this.p.push([this.x, this.y - i])
+              }
+              this.y -= num
+          }
+        return this
+    }
+
+    right(){
+        if(this.way === 'east'){
+            this.way = 'south';
+        } else if(this.way === 'south'){
+            this.way = 'west';
+        } else if(this.way === 'west'){
+            this.way = 'north';
+        } else if(this.way === 'north'){
+            this.way = 'east';
+        }
+        return this
+    }
+
+    left(){
+        if(this.way === 'east'){
+            this.way = 'north'
+        } else if(this.way === 'north'){
+            this.way = 'west'
+        } else if(this.way === 'west'){
+            this.way = 'south'
+        } else if(this.way === 'south'){
+            this.way = 'east'
+        }
+        return this
+    }
+
+    allPoints(){
+        return this        
+    }
+
+    print(){
+        this.xMax = 0;
+        this.yMax = 0;
+
+        for (let i = 0; i < this.p.length; i++){
+            let xVal = this.p[i][0];
+            let yVal = this.p[i][1];
+
+            if (xVal > this.xMax){
+                this.xMax = xVal;                
+            }
+            
+            if (yVal > this.yMax){
+                this.yMax = yVal;
+            }           
+        };
+        
+        this.max = [this.xMax, this.yMax];
+        //console.log(this.max); 
+
+        let draw = "-- BEGIN LOG" +"\n";
+
+        for (let i = 0; i <= this.max[1]; i++){
```

Comment:
it's not much of a big deal at this point but avoid using nested for loops. bad performance in real practice.